src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java

```diff
@@ -69,12 +78,80 @@ public Status deleteEvent() throws SQLException {
 		mEvents.delete(mEventId);
 		return Status.SUCCESS_OK;
 	}
-	
+
 	private static class UpdateEventEntity {
 		public String name, description;
 		public boolean current, reserved_to_affiliates, reserved_to_partners;
 		public int minimum_views, minimum_followers;
 		public EventBean.Status status;
 		public EventBean.RankingType ranking_type;
 	}
+
+	private void rankAllUsers(int eventId) throws SQLException {
+		EventBean event = mEvents.getEvent(mEventId);
+		List<EventsDao.RoundUserScoreBean> allScores = mEvents.getAllScores(eventId);
+
+		Set<Integer> lastUsers = new HashSet<>();
+
+		Map<Integer,Double> map = new HashMap<>();
+
+		for (EventsDao.RoundUserScoreBean rus:
+				allScores) {
+			if (rus.getScore() == 0){
+				lastUsers.add(rus.getId());
+			}
+			if (map.containsKey(rus.getId())){
+				map.put(rus.getId(),map.get(rus.getId()) + rus.getScore());
+			} else {
+				map.put(rus.getId(),rus.getScore());
+			}
+		}
+
+		for (Integer userId:
+			 lastUsers) {
+			map.remove(userId);
+		}
+
+		List<Map.Entry<Integer,Double>> list = new ArrayList<>(map.entrySet());
+		Collections.sort(list, new ScoreUserValueComparator(event.getRankingType()));
+
+		int i = 1;
+		int currentRank = 1;
+		Double currentScore = null ;
+		for (Map.Entry<Integer,Double> entry:
+				list) {
+
+			if (currentScore == null || !currentScore.equals(entry.getValue())){
+				currentRank = i;
+				currentScore = entry.getValue();
+			}
+
+			mEvents.updateUserRank(mEventId, entry.getKey(), currentRank);
+
+			i++;
+		}
+
+		for (Integer userId:
+			 lastUsers) {
+			mEvents.updateUserRank(mEventId, userId, i);
+		}
+
+	}
+
+	class ScoreUserValueComparator implements Comparator<Map.Entry<Integer, Double>> {
+		EventBean.RankingType rankingType;
```

Comment:
This should be private final

Reply:
Also, when this is private final, you can remove constructor and use lombok `@RequiredArgsConstructor`